concordium-contracts-common/src/types.rs

```diff
@@ -1851,6 +1848,24 @@ mod serde_impl {
         use super::*;
         use rand::Rng;
 
+        #[test]
+        fn test_account_address_to_string_parse_is_id() {
+            use rand::Rng;
+            let mut rng = rand::thread_rng();
+            let mut address_bytes = [0u8; 32];
+
+            for _ in 0..1000 {
+                let address = AccountAddress(address_bytes);
+                rng.fill(&mut address_bytes);
```

Comment:
I believe these two lines should be swapped. The byte array will be *copied* into the account address.